rlog/2024-12-08-merkle-tree.mdx

```diff
@@ -0,0 +1,197 @@
+---
+title: 'Vac 101 - Climbing Merkle Trees'
+date: 2024-12-08 12:00:00
+authors: marvin
+published: true
+slug: climbing-merkle-trees
+categories: research
+
+toc_min_heading_level: 2
+toc_max_heading_level: 4
+---
+
+In this post, we introduce a crucial data structure used throughout web3.
+
+<!--truncate-->
+
+## Introduction
+
+A large amount of data is swapped between users on a blockchain in the form of transactions.
+Over the entire life of a blockchain,
+the storage space required to maintain a copy of every transaction becomes untenable for most users.
+However, the integrity of a blockchain relies on a large pool of users that can validate the blockchain's history from its inception to its present state.
+The data representing the blockchain's state is compressed.
+This compression addresses the issue of scale that would greatly restrict the pool of users.
+
+Data compression alone is not the end goal.
+As mentioned, it is essential for users to beable to validate the blockchain's history.
+The property of compression and validation was solved in Bitcoin by the use of Merkle trees.
```

Comment:
We can mention some use case examples, such as nullifying, proving membership (e.g. RLN), lightclients.